### ISSUE TITLE
[RL] Unify mask creation logic for RL.

### ISSUE THREAD

**Issue by zhxchen17** [pytorch/torchtitan#2716]
Today on trainer we create attention mask using the helper function.

We should investigate whether this piece can be merged and put into a dedicated place with code from https://github.com/verl-project/verl/pull/5051/changes#diff-d90b34d27a9c17f5a898247fd8858ab3e61876d67ed2d8653e18bd5f85658a67R156 (`_create_varlen_metadata_for_document`).

Context: following up https://github.com/pytorch/torchtitan/pull/2649

cc @acisseJZhong

**Comment by acisseJZhong** [comment]
cc @joecummings
